Documentation/ITK5MigrationGuide.md

```diff
@@ -459,6 +459,117 @@ expanded, but previous template parameters may not be available.
 
 Arguments to functions in `itkExtras` were cleaned up and now use snake case.
 
+Strongly Typed Enumerations
+---------------------------
+
+In ITK 5, enumerations are strongly typed, declared with `enum class`. A best
+practice in modern C++, strongly typed enum's [offer the following improvements](https://www.modernescpp.com/index.php/strongly-typed-enums):
+
+- The enumerators can only be accessed in the scope of the enumeration.
+- The enumerators don't implicitly convert to `int`.
+- The enumerators aren't imported in the enclosing scope.
+- The type of the enumerators is by default `int`. Therefore, you can forward the enumeration.
+
+In order to preserve as much backwards compatibility as possible, appropriate
+scoping, provide clean, readable code, facilitate wrapping in languages such
+as Python, and enable printing enum values to `std::ostream` with
+`operator<<`, and support templates, enums that we previously declared as:
+
+```
+// itkClassName.h
+namespace itk
+{
+
+class ClassName
+{
+public:
+
+  enum Choices
+  {
+     One,
+     Two,
+     Three
+  };
+};
+
+}
+```
+
+are now declared as:
+
+```
+// itkClassName.h
+namespace itk
+{
+
+class ClassNameEnums
```

Comment:
Should be the same as name on line 518

Reply:
It has to be different for the case where `ClassName` is templated and for Python wrapping.